### PR TITLE
Issue/7601 media video settings

### LINKED ISSUE #7601: Implement Video size settings on App Settings
### Expected behavior

On App settings we should have an option to tweak the size of the video being uploaded to the site.

This can be very important for users with low bandwidth our small data plans.

### Actual behavior

We only allow to setup the size of images not videos

### Steps to reproduce the behavior

Go to Me -> App Settings

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #7601 

To test:
 - Go to Me ->App Settings
 - Change the Video Size settings
 - Check if going in and out the setting is correctly saved
 - Go to Blog -> Posts
 - Create a new Post
 - Add a video to the post
 - Check that the video export process goes well
 - Check in the server if the video respected the maximum upload video size
 - Remember the settings doesn't upscale videos!

### CODE REVIEW

File: WordPress/Classes/Services/MediaSettings.swift [hunk 1]
```diff
@@ -1,13 +1,95 @@
...
+                return AVAssetExportPreset3840x2160
+            case(.sizeOriginal):
+                return AVAssetExportPresetPassthrough
+            }
+        }
+
+        var description: String {
+            switch (self) {
+            case(.size640x480):
+                return NSLocalizedString("VGA (640x480)", comment: "Indicates a video will be resized to 640x480 when uploaded.")
```
Comment (frosty) [inline WordPress/Classes/Services/MediaSettings.swift]: I wonder if we could also ditch 540p – it's only marginally more than 480p, and it's less common than either 480 or 720.

File: WordPress/Classes/Services/MediaSettings.swift [hunk 2]
```diff
@@ -1,13 +1,95 @@
...
+                return NSLocalizedString("Full HD (1920x1080)", comment: "Indicates a video will be resized to Full HD 1920x1080 when uploaded.")
+            case(.size3840x2160):
+                return NSLocalizedString("4K(3840x2160)", comment: "Indicates a video will be resized to 4K 3840x2160 when uploaded.")
+            case(.sizeOriginal):
+                return NSLocalizedString("Original", comment: "Indicates a video will use its original size when uploaded.")
+            }
+        }
+
+        var intValue: Int {
+            switch (self) {
```
Comment (frosty) [inline WordPress/Classes/Services/MediaSettings.swift]: I don't really like the repetition of these switch cases switching on or returning literal numbers of 1, 2, 3, 4, 5, 6 etc. But I can't think of many better ways of doing it. Perhaps to make it clearer, we could just move these into two methods on `VideoSizingRow`: `sliderValueForResolution`, and `resolutionForSliderValue` or something?

Comment (frosty) [inline WordPress/Classes/Services/MediaSettings.swift]: One other option, but I'm not convinced I like it any better: https://gist.github.com/anonymous/6b83ce5fbee4c022b1efe8a36cef6d46

File: WordPress/Classes/ViewRelated/Me/AppSettingsViewController.swift [hunk 3]
```diff
@@ -350,6 +362,81 @@ fileprivate struct MediaSizingRow: ImmuTableRow {
...
+        let cell = cell as! CellType
+        cell.model = VideoSizeModel.default
+        cell.title = title
+        cell.value = value.intValue
+        cell.onChange = { (value) in
+            self.onChange(MediaSettings.VideoResolution.videoResolution(from: value))
+        }
+        cell.selectionStyle = .none
+
+        cell.minValue = MediaSettings.VideoResolution.size640x480.intValue
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Me/AppSettingsViewController.swift]: Let's use `VideoSizeModel.default.minValue` and `VideoSizeModel.default.maxValue` here.

**Comment by frosty** [comment]
cc @kwonye, as this is relevant to your interests

File: WordPress/Classes/Services/MediaSettings.swift (hunk 1 above)
Comment (SergioEstevao) [inline WordPress/Classes/Services/MediaSettings.swift]: Agree! I like those name better, and I will ditch the 540p.

File: WordPress/Classes/Services/MediaSettings.swift [hunk 4]
```diff
@@ -1,13 +1,95 @@
...
+        case size640x480 = "AVAssetExportPreset640x480"
+        case size960x540 = "AVAssetExportPreset960x540"
+        case size1280x720 = "AVAssetExportPreset1280x720"
+        case size1920x1080 = "AVAssetExportPreset1920x1080"
+        case size3840x2160 = "AVAssetExportPreset3840x2160"
+        case sizeOriginal = "AVAssetExportPresetPassthrough"
+
+        var videoPreset: String {
+            switch (self) {
+            case(.size640x480):
```
Comment (SergioEstevao) [inline WordPress/Classes/Services/MediaSettings.swift]: I blame the Objective-C in me for those :).

File: WordPress/Classes/Services/MediaSettings.swift (hunk 2 above)
Comment (SergioEstevao) [inline WordPress/Classes/Services/MediaSettings.swift]: Humm, I don't think it makes any better... do you think it's worth the change?

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

**Comment by SergioEstevao** [comment]
@frosty I removed the 540p option.

File: WordPress/Classes/Services/MediaSettings.swift [hunk 5]
```diff
@@ -57,8 +52,6 @@ class MediaSettings: NSObject {
             switch self {
             case .size640x480:
                 return 1
-            case .size960x540:
-                return 2
```
Comment (frosty) [inline WordPress/Classes/Services/MediaSettings.swift]: Can we update the other numbers here so they're consecutive?

**Comment by SergioEstevao** [comment]
@frosty Good catch! just update the code, ready for another look.

**Comment by designsimply** [comment]
Tested and confirmed using the following testing steps:

1. Go to Me > App Settings > Media and change "Max Video Upload Size" to 480p.
1. Create and publish a new post with a video added.
1. Check the video details via the Media Library and compare them to the original video

<strong>Result:</strong> for my test, the uploaded video was 13MB at 640 x 360 while the original video shot with my iPhone 6S was 68MB at 1920 × 1080.

img_1367 img_1368

**Comment by kwonye** [comment]
@frosty is this a situation where the aspect ratio is different hence the incorrect height?

**Pull request merged**